subsys/storage/flash_map/flash_map.c

```diff
@@ -270,6 +270,39 @@ const struct device *flash_area_get_device(const struct flash_area *fa)
 	return device_get_binding(fa->fa_dev_name);
 }
 
+uint8_t flash_area_erased_val(const struct flash_area *fa)
+{
+	const struct flash_parameters *param;
+
+	param = flash_get_parameters(device_get_binding(fa->fa_dev_name));
+
+	return param->erase_value;
+}
+
+int flash_area_read_is_empty(const struct flash_area *fa, off_t off,
+			     void *dst, size_t len)
+{
+	size_t i;
+	int rc;
+	uint8_t *u8dst;
+	uint8_t erased_val;
+
+	rc = flash_area_read(fa, off, dst, len);
+	if (rc) {
+		return -1;
+	}
```

Comment:
I think that this function may return 1 (memory is erased) in case when `len` is given 0. `len == 0` is unusual situation, how should we handle it in this case?

Reply:
Let's be more precise on that, `len == 0` imposes returns of `-EINVAL`.